### ISSUE TITLE
An illegal reflective access operation has occurred Java 11

### ISSUE THREAD

**Issue by khmarbaise** [jqno/equalsverifier#248]
# What steps will reproduce the problem?
Creating a class:
```java
package x;

import java.lang.reflect.Method;
import java.util.Objects;

public class FieldA {

  private final Method x;

  public FieldA(Method x) {
    this.x = x;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof FieldA)) {
      return false;
    }
    FieldA fieldA = (FieldA) o;
    return Objects.equals(x, fieldA.x);
  }

  @Override
  public int hashCode() {
    return Objects.hash(x);
  }
}
```
and creating a test:
```java
package x;

import nl.jqno.equalsverifier.EqualsVerifier;
import org.junit.jupiter.api.Test;

class FieldATest {

  @Test
  void equalsVerifier() {
    EqualsVerifier.forClass(FieldA.class).verify();
  }
}
```

# What is the code that triggers this problem?
Complete example above.

# What error message or stack trace does EqualsVerifier give?
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by nl.jqno.equalsverifier.internal.reflection.FieldAccessor (file:/Users/user/.m2/repository/nl/jqno/equalsverifier/equalsverifier/3.1.9/equalsverifier-3.1.9.jar) to field sun.reflect.generics.repository.MethodRepository.returnType
WARNING: Please consider reporting this to the maintainers of nl.jqno.equalsverifier.internal.reflection.FieldAccessor
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release


java.lang.AssertionError: EqualsVerifier found a problem in class FieldA.
-> java.lang.reflect.InvocationTargetException
```

# What did you expect?

# Which version of EqualsVerifier are you using?
Version 3.1.9
# Please provide any additional information below.
If you need further informations don't hesitate to contact me..

**Comment by jqno** [comment]
Thanks for reporting this. I've reproduced the issue, I'll try to fix it soon.

**Comment by jqno** [comment]
Should be fixed in version 3.1.10.

**Comment by khmarbaise** [comment]
Checked with 3.1.10 and works now perfect. Thanks for that fast fix...
